In [1]:
from funcoes import valida_opcoes, obter_opcoes

In [5]:
import json

In [28]:
def carrega_dados(path:str = 'professores.json') -> list[dict]:
    try:
        with open(path, 'r') as arquivo:
            dados = arquivo.read()
            return json.loads(dados)
    except FileNotFoundError:
        return []
    
def salvar_dados(dados:list[dict], path:str = 'professores.json') -> bool:
    try:
        with open(path, 'w') as arquivo:
            arquivo.write(json.dumps(dados))
            return True
    except Exception:
        return False
    
def formata_dados_professor(prof: dict) -> str:
    return '\n'.join([f'{key.title()}: {value}' for key, value in prof.items()])

def formata_dados_todos_professores(profs: list[dict]) -> str:
    return '\n\n'.join(list(map(formata_dados_professor, profs)))


In [29]:
print(formata_dados_todos_professores(dados))

Nome : Roberto
Idade : 35
Sexo : Masculino
Altura : 1.75
Disciplina : Matemática

Nome : Professor2
Idade : 42
Sexo : Feminino
Altura : 1.68
Disciplina : História

Nome : Professor3
Idade : 38
Sexo : Masculino
Altura : 1.8
Disciplina : Ciências

Nome : Professor4
Idade : 29
Sexo : Feminino
Altura : 1.65
Disciplina : Inglês

Nome : Professor5
Idade : 45
Sexo : Masculino
Altura : 1.72
Disciplina : Educação Física


{'nome': 'Roberto',
  'idade': 35,
  'sexo': 'Masculino',
  'altura': 1.75,
  'disciplina': 'Matemática'}

In [19]:
dados

[{'nome': 'Roberto',
  'idade': 35,
  'sexo': 'Masculino',
  'altura': 1.75,
  'disciplina': 'Matemática'},
 {'nome': 'Professor2',
  'idade': 42,
  'sexo': 'Feminino',
  'altura': 1.68,
  'disciplina': 'História'},
 {'nome': 'Professor3',
  'idade': 38,
  'sexo': 'Masculino',
  'altura': 1.8,
  'disciplina': 'Ciências'},
 {'nome': 'Professor4',
  'idade': 29,
  'sexo': 'Feminino',
  'altura': 1.65,
  'disciplina': 'Inglês'},
 {'nome': 'Professor5',
  'idade': 45,
  'sexo': 'Masculino',
  'altura': 1.72,
  'disciplina': 'Educação Física'}]

In [13]:
dados = carrega_dados()

In [18]:
dados

[{'nome': 'Roberto',
  'idade': 35,
  'sexo': 'Masculino',
  'altura': 1.75,
  'disciplina': 'Matemática'},
 {'nome': 'Professor2',
  'idade': 42,
  'sexo': 'Feminino',
  'altura': 1.68,
  'disciplina': 'História'},
 {'nome': 'Professor3',
  'idade': 38,
  'sexo': 'Masculino',
  'altura': 1.8,
  'disciplina': 'Ciências'},
 {'nome': 'Professor4',
  'idade': 29,
  'sexo': 'Feminino',
  'altura': 1.65,
  'disciplina': 'Inglês'},
 {'nome': 'Professor5',
  'idade': 45,
  'sexo': 'Masculino',
  'altura': 1.72,
  'disciplina': 'Educação Física'}]

In [ ]:
#TODO: Criar uma Função Pura

def inserir_novo(profs: list[dict], **kwargs) -> None:
    profs.append(kwargs)
    

In [3]:
opc = {
    'I': 'Inserir', #salvando dados 
    'ET': 'Exibir Todos'
}

In [4]:
obter_opcoes(opc, 'Escolha uma ação')

'I'

In [ ]:
while True:
    
    
    
        
    if obter_opcoes({'S': 'Sim', 'N': 'Não'}, 'Deseja Sair') == 'S':
        break